File: javabackend/src/main/java/mft/dev/javabackend/repository/TodoRepository.java
package mft.dev.javabackend.repository;

import mft.dev.javabackend.model.Todo;
import mft.dev.javabackend.model.Utente;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface TodoRepository extends JpaRepository<Todo, Long> {

    List<Todo> findByUtente(Utente utente);

    void deleteAllByUtente(Utente utente);
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/repository/TodoRepository.kt
package mft.dev.kotlinbackend.repository

import mft.dev.kotlinbackend.model.Todo
import mft.dev.kotlinbackend.model.Utente
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TodoRepository : JpaRepository<Todo, Long> {
    fun findByUtente(utente: Utente): MutableList<Todo>

    fun deleteByUtente(utente: Utente)
}